packages/compass-web/src/logger-and-telemetry.tsx

```diff
@@ -0,0 +1,187 @@
+import type {
+  LoggerAndTelemetry,
+  MongoLogWriter,
+} from '@mongodb-js/compass-logging/provider';
+import { Writable } from 'stream';
+import { mongoLogId } from '@mongodb-js/compass-logging/provider';
+import { useRef } from 'react';
+
+export type TrackFunction = (
+  event: string,
+  properties: Record<string, any>
+) => void;
+
+export type LogFunction = (
+  type: 'debug' | 'info' | 'warn' | 'error' | 'fatal',
+  ...args: any[]
+) => void;
+
+export type DebugFunction = (...args: any[]) => void;
+
+class CompassWebLogWriter extends Writable implements MongoLogWriter {
+  _logId = '';
+  get logId() {
+    return this._logId;
+  }
+  _logFilePath = null;
+  get logFilePath() {
+    return this._logFilePath;
+  }
+  _target = {} as Writable;
+  get target() {
+    return this._target;
+  }
+  _now = () => {
+    return new Date();
+  };
```

Comment:
Required on the MongoLogWriter interface that LoggerAndTelemetry expects, but not really used by anything, so we're providing a bunch of no-ops here